### ISSUE TITLE
Scalatest output

### ISSUE THREAD

**Issue by cranst0n** [tpolecat/doobie#406]
[This](https://github.com/tpolecat/doobie/blob/39804b154ee4958e1c4e191c00f1c41e21290606/yax/scalatest/src/main/scala/doobie/scalatest/QueryChecker.scala#L78) seems like a bug where no output is printed on a successful test.

I've modified it in my own project to this which seems to work:
```scala
val analysisOutput = transactor.trans(analysis).attempt.unsafePerformIO match {
  case -\/(e) =>
    failure("SQL fails typechecking", formatError(e.getMessage))
  case \/-(a) =>
    success("SQL compiles and typechecks", None) +
      a.paramDescriptions.map { case (s, es) => assertEmpty(s, es) }
        .map(s => s"  $s")
        .mkString +
      a.columnDescriptions.map { case (s, es) => assertEmpty(s, es) }
        .map(s => s"  $s")
        .mkString
}

println(
  s"  $typeName defined at ${loc(pos)}\n" +
    formatSql(sql) +
    analysisOutput
)
```

**Comment by tpolecat** [comment]
Yeah I wonder if I should change the specs2 one to be silent on success. It is kind of noisy.

**Comment by cranst0n** [comment]
Personal preference I guess but I think no output is confusing.

At a minimum I would like to see the output of the [success](https://github.com/tpolecat/doobie/blob/39804b154ee4958e1c4e191c00f1c41e21290606/yax/scalatest/src/main/scala/doobie/scalatest/QueryChecker.scala#L125) function from a good test. I don't think I'd miss the column/parameter descriptions on a passing test but other people may have stronger opinions.

**Comment by tpolecat** [comment]
You can pull the PR above if you want to see how it looks after my tweaks. There were a few issues. I can't seen an obvious way to do output on success so I'm actually not fixing this particular issue this time around … the Specs2 mixin turns each check into a set of testcases, each of which has independent success/failure and it's all reported in a nice way, but the ScalaTest mixin does each check as a single test. It could probably be done the same way I did Specs2 but I don't have time to rewrite it before 0.4.

So we can leave this open and revisit.

**Comment by cranst0n** [comment]
Gotcha. I've got my own variation for the time being that just uses a `println` in the success case so I can sleep at night.

I'm full up right now but if I can find time, and you're interested, I can try to come up with something like you're describing based on the specs2 implementation for a PR.
